react/features/invite/functions.ts

```diff
@@ -203,14 +244,32 @@ export function getInviteResultsForQuery(
                 = peopleResults.find(result => result.type === INVITE_TYPES.PHONE);
 
             if (!hasPhoneResult && typeof phoneResults.allow === 'boolean') {
-                results.push({
+                let result = {
                     allowed: phoneResults.allow,
                     country: phoneResults.country,
                     type: INVITE_TYPES.PHONE,
                     number: phoneResults.phone,
                     originalEntry: text,
                     showCountryCodeReminder: !hasCountryCode
-                });
+                };
+
+                if (phoneResults.allow) {
+                    results.push(result);
+                } else {
+                    checkOutboundDestination(dialOutDestUrl, jwt, appId, text)
```

Comment:
you need to use async await, otherwise `return results` will be called before adding this result to the results array, as the execution will not wait for the request to finish.